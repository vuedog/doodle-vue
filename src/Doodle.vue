<template>
  <div>
    <p>{{ template }}</p>
  </div>  
</template>

<script>
export default {
  props: ['msg'],
  data () {
    return {template: this.msg || 'default msg'}
  }
}
</script>

