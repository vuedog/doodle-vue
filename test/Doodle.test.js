import { shallow, mount } from 'vue-test-utils'
import Doodle from '@/Doodle'
import { createRenderer } from 'vue-server-renderer'

describe('Doodle', () => {
  it('renders props.msg when passed', () => {
    const msg = 'new message'
    const wrapper = mount(Doodle, {
      propsData: {
        msg
      }
    })
    expect(wrapper.html()).toBe('<div><p>new message</p></div>')
  })
  it('matches snapshot', () => {
    const wrapper = shallow(Doodle, {})
    const renderer = createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
