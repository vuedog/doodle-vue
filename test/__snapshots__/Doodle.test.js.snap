// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doodle matches snapshot 1`] = `
<div>
    <p>default msg</p>
</div>
`;
